### PR TITLE
br: modify to use one kubeconfig for multiple data planes

### LINKED ISSUE #5183: Change to use one kubeconfig configuration in br federation manager
## Feature Request

**Describe the feature you'd like:**

Now, the credentials used to authenticate data planes should be separated to multiple kubeconfig files. Since the kubeconfig file  supports contexts, one kubeconfig file is enough. So we can change to use one kubeconfig file.

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

Modify to use one kubeconfig for multiple data planes

Closes #5183 

### What is changed and how does it work?

Change the flag `federation-kubeconfig-path` to the kubeconfig file path. Then read the kubeconfig file and use the contexts in kubeconfig file mapping to the data planes

After this change, the secret that stores the kubeconfig file should be configured like this:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: br-federation-kubeconfig
type: Opaque
data:
  kubeconfig: {encoded-kubeconfig}
```

If the user want to use custom secret name or custom data key, he should change the value of chart `.brFederationManager.federationKubeconfigSecret` or `.brFederationManager.federationKubeconfigSecretKey`

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: charts/br-federation/values.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ brFederationManager:
   # Secret name of the kubeconfig for the federation Kubernetes clusters
   # The data item key is the cluster name, and the data item value is the base64 encoded kubeconfig
   federationKubeconfigSecret: br-federation-kubeconfig
+  federationKubeconfigSecretKey: kubeconfig
```
Comment (BornChanger) [inline charts/br-federation/values.yaml:25]: Add some description to the new field

**Review by BornChanger — APPROVED** [review]

File: charts/br-federation/values.yaml (hunk 1 above)
Comment (WangLe1321) [reply charts/br-federation/values.yaml:27]: done

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**